_studio/shared/umc/core/umc/include/umc_structures.h

```diff
@@ -691,7 +691,7 @@ namespace UMC
     {
         // some compilers complain to conversion to/from
         // pointer types from/to integral types.
-        return (T) ((((uint8_t *) pv - (uint8_t *) 0) + (lAlignValue - 1)) &
```

Comment:
Why do we have changes here? Please explain.

Reply:
This change was introduced to fix below build error:

Clang-14 complains about an unusual way of converting
a pointer to an integer:

error: invalid operands to binary expression
('uintptr_t *' (aka 'unsigned long *') and 'unsigned long')
   return (T) ((((uintptr_t *) pv) + (lAlignValue - 1)) &

Replace this with a normal cast to uintptr_t

Reply:
I created the patch based on this patch: https://www.spinics.net/lists/stable-commits/msg218632.html

Reply:
okay, thanks for the explanation.